# Whole Genome methylation Association Analysis for Allergic Asthma (AA):  Aim 1


## Summary 

This report documents the results of whole genome methylation association analysis for allergic asthma (AA) under Aim 1. 


## Data 

Phenotype and genotype of interests:

* AA only versus control (neither AA nor AD)

* Methylation profile (methylation proportions at each CpG; after arc-sin square root transformation) from  Illumina 450k microarray.

Other Covariates:

* age at recruitment (AgeCalc)
* sex
* smoking status. Non-smoker versus smoker (current or former).
* Five surrogate variables (SVs) to capture hidden sources of variation in the methylation data. sv1, …sv5.
* Cell-type proportions.
* Random Effect: Family ID (FID) was included as a random effect to account for shared genetic and environmental factors within families.


## Models 


### Model 1

* Y: AA Status (0/1)

* X: arc-sin square root transformed methylation beta values at each CpG. 

* Other covariates as fixed effects.

* Family ID (FID) as random effect.

* algorithm: glmmLASSO with logit link.


* pros: 
  * 1. The fixed effects will be forced toward zero via optimized regularization, therefore the collinearity issue is resolved.
  * 2. The glmmLASSO algorithm is computationally efficient comparing to other penalization methods, making it suitable for whole genome analysis.

* cons:
    * 1. The choice of penalty parameter lambda can affect the results, and there is no universal method for selecting it. In our practise we select the lambda value by optimizing the model fit according to BIC criteria.
    * 3. Interpretation of the results can be challenging due to the complexity of the model.



### Model 2

* Y: arc-sin square root transformed methylation beta values at each CpG. 

* X: AA Status (0/1)  

* Other covariates as fixed effects.

* Family ID (FID) as random effect.

* algorithm: linear mixed effect model.


* pros: 
    * 1. The model is easy to interpret, as the coefficients represent the change in methylation associated with AA status.

* cons:
    * 1. The model assumes that the residuals are normally distributed and homoscedastic, which may not be universally true for all CpGs.
    * 2. An example is we observed that the model 2 will fail at CpGs where most methylation proportions are close to 0. We have to ignore these CpGs for model 2 analysis.
    * 3. The model can lead to overfitting and false positives for feature selection, since there are 13 fixed effects and 1 random effect in the model.
    * 4. The output indicated that for many CpGs the random effect for family had near-zero variance, suggesting little variability between families at those sites. In fact, around 10% of the top 10,733 CpGs are missing random effects! We have to ignore these CpGs for whole genome analysis.



### Model 3

* Y: arc-sin square root transformed methylation beta values at each CpG. 

* X: AA Status (0/1) 

* Other covariates as fixed effects.

* Family ID (FID) as random effect.

* algorithm: glmmLASSO with logit link.


* pros: 
    * 1. The model is easy to interpret, as the coefficients represent the change in methylation associated with AA status.
    * 2. The model is computationally efficient comparing to other penalization methods, making it suitable for whole genome analysis.

* cons:
    * 1. The choice of penalty parameter lambda can affect the results, and there is no universal method for selecting it. In our practise we select the lambda value by optimizing the model fit according to BIC criteria.
    * 2. Interpretation of the results can be challenging due to the complexity of the model.
    * 3. The model assumes that the residuals are normally distributed and homoscedastic, which may not be universally true for all CpGs.



    



## Results 


### Manhattan Plots

We demonstrate the Manhattan plots of methylation association pvalues from Model 1-3. The x-axis is the genomic position of each CpG site, and the y-axis is the -log10(p-value) of the association between methylation and AA status. The horizental line indicates the genome-wide significance threshold (5e-8 and 1e-5). 

In summary: 

* Model 1 have 58 CpGs with pvalue < 1e-5.
* Model 2 have 90 CpGs with pvalue < 1e-5.
* Model 3 have 0 CpGs with pvalue < 1e-5.



![Figure 1. Manhattan plot for pvalues from model 1. ](M1_manhattan.jpg)


![Figure 2. Manhattan plot for pvalues from model 2. ](M2_manhattan.jpg)

![Figure 3. Manhattan plot for pvalues from model 3. ](M3_manhattan.jpg)






## Agreement between models


Vendiagram of the signifcant CpGs from model 1 and model 2 under threshold pvalue < 1e-5.

![Figure 4. Vendiagram of significant CpGs from Model 1 and Model 2 under threshold pvalue < 1e-5. ](venn_12.jpeg)




## Annotated genes 



### Model 1: 



58 CpGs with pvalue < 1e-5.




In [7]:
import pandas as pd

# 1) load your CSV into a DataFrame
df = pd.read_csv("M1_cpgs_genes.csv")

# 2) just put `df` as the last line in a cell to get the pretty notebook table
df


,CpG,genes_entrez,genes_symbol
0,10:119303306-119303307,196047;2018,EMX2OS;EMX2
1,10:99474180-99474181,83742,MARVELD1
2,11:121028586-121028587,7007,TECTA
3,11:47207027-47207028,29763,PACSIN3
4,11:62345261-62345262,64852,TUT1
5,12:133382384-133382385,2802,GOLGA3
6,14:105860323-105860324,23241,PACS2
7,14:76958030-76958031,2103,ESRRB
8,16:57567738-57567739,92922,CCDC102A
9,17:17465320-17465321,10400,PEMT


* PDCD1LG2 (PD-L2)

Multiple in-vivo and human‐biopsy studies (A K Singh, et al 2010; IP Lewkowich, et al, 2014) show that PD-L2 expression on pulmonary dendritic cells promotes airway hyperresponsiveness in allergic asthma. Blockade of PD-L2 in mouse models of house-dust-mite or ovalbumin-induced asthma significantly reduces bronchoconstriction and mucus production—without altering Th2‐cytokine levels—by restoring IL-12 production and dampening IL-13 signaling.

* PROX1 

PROX1 is a master regulator of lymphatic-endothelial identity (i.e. Prox-1⁺ LYVE-1⁺ cells), and several studies have used Prox-1–GFP mice to visualize lymphatic remodeling in allergen-induced asthma. 



Functional enrichment analysis of the top 1000 CpGs from Model 1 : 


In [8]:
import pandas as pd

# 1) load your CSV into a DataFrame
df = pd.read_csv("M1_GO.csv")

# 2) just put `df` as the last line in a cell to get the pretty notebook table
df

,ID,Description,GeneRatio,BgRatio,RichFactor,FoldEnrichment,zScore,pvalue,p.adjust,qvalue,geneID,Count
0,GO:0007156,homophilic cell adhesion via plasma membrane a...,21/448,168/18986,0.125000,5.297433,8.697409,4.983010e-10,0.000002,0.000002,RET/CDH23/PKD1/CDH5/CELSR1/CLSTN2/CNTN4/ROBO1/...,21
1,GO:0030900,forebrain development,29/448,416/18986,0.069712,2.954338,6.265463,2.228232e-07,0.000260,0.000234,EMX2/NRP1/ZMIZ1/TH/SLC1A2/LHX5/SYNE2/FOS/BCL11...,29
2,GO:0098742,cell-cell adhesion via plasma-membrane adhesio...,23/448,280/18986,0.082143,3.481170,6.502165,2.373494e-07,0.000260,0.000234,RET/CDH23/PKD1/CDH5/CELSR1/CLSTN2/IL1RAP/CNTN4...,23
3,GO:0021537,telencephalon development,23/448,281/18986,0.081851,3.468782,6.481416,2.529541e-07,0.000260,0.000234,EMX2/ZMIZ1/TH/SLC1A2/LHX5/SYNE2/FOS/BCL11B/IGF...,23
4,GO:0060840,artery development,14/448,116/18986,0.120690,5.114763,6.910361,6.120823e-07,0.000504,0.000452,NRP1/EGR2/ZMIZ1/WNT11/ARID2/LRP1/SMAD7/COMP/PR...,14
5,GO:0021543,pallium development,18/448,195/18986,0.092308,3.911951,6.353883,9.147618e-07,0.000627,0.000563,EMX2/ZMIZ1/TH/LHX5/SYNE2/FOS/IGF1R/NDE1/FXR2/P...,18
6,GO:0048844,artery morphogenesis,11/448,83/18986,0.132530,5.616556,6.552463,3.911001e-06,0.002299,0.002064,NRP1/ZMIZ1/WNT11/ARID2/LRP1/SMAD7/COMP/PRRX1/P...,11
7,GO:0021987,cerebral cortex development,13/448,130/18986,0.100000,4.237946,5.758766,1.263103e-05,0.006166,0.005535,EMX2/ZMIZ1/TH/SYNE2/FOS/NDE1/CDK5R2/ROBO1/POU3...,13
8,GO:0048568,embryonic organ development,27/448,458/18986,0.058952,2.498353,5.045978,1.348625e-05,0.006166,0.005535,CDH23/TECTA/IGF2/TH/KCNQ1/WNT11/ARID2/HOXC4/AT...,27
9,GO:0060041,retina development in camera-type eye,15/448,174/18986,0.086207,3.653402,5.466043,1.684439e-05,0.006931,0.006222,NRP1/RET/TCIRG1/LRP5/ATP8A2/FOS/NR2E3/STAT3/RD...,15


Among the top 3 enriched GO BP terms, the two adhesion‐related categories —

GO:0007156 “homophilic cell adhesion via plasma membrane adhesion molecules”

GO:0098742 “cell–cell adhesion via plasma‐membrane adhesion molecules”

— are related to allergic asthma. In asthma, adhesive interactions between immune cells and airway structural cells drive inflammation and remodeling.

### Model 2: 



90 CpGs with pvalue < 1e-5.

In [9]:
import pandas as pd

# 1) load your CSV into a DataFrame
df = pd.read_csv("M2_cpgs_genes.csv")

# 2) just put `df` as the last line in a cell to get the pretty notebook table
df

,CpG,genes_entrez,genes_symbol
0,10:119303306-119303307,196047;2018,EMX2OS;EMX2
1,10:82300841-82300842,387694,SH2D4B
2,10:99474180-99474181,83742,MARVELD1
3,11:10830493-10830494,1982,EIF4G2
4,11:121028536-NA,7007,TECTA
5,11:121028586-121028587,7007,TECTA
6,11:121028714-121028715,7007,TECTA
7,11:47207027-47207028,29763,PACSIN3
8,12:133382384-133382385,2802,GOLGA3
9,13:113446003-NA,23250,ATP11A


None of these genes are really interesting, except PRDM16, which was found with GWAS signals from an Arican-American GWAS study.

In [10]:
import pandas as pd
import ipywidgets as widgets
from IPython.display import display

# 1) load your CSV into a DataFrame
df = pd.read_csv("M2_GO.csv")


display(df)


# 2) just put `df` as the last line in a cell to get the pretty notebook table


,ID,Description,GeneRatio,BgRatio,RichFactor,FoldEnrichment,zScore,pvalue,p.adjust,qvalue,geneID,Count
0,GO:0007409,axonogenesis,34/478,459/18986,0.074074,2.942197,6.769189,2.174272e-08,0.000086,0.000076,RET/ANK3/EGR2/PTEN/EIF4G2/NCAM1/CTTN/ULK1/CYFI...,34
1,GO:0021537,telencephalon development,25/478,281/18986,0.088968,3.533778,6.876736,5.206689e-08,0.000086,0.000076,EMX2/ZMIZ1/PTEN/TH/SLC1A2/LHX5/SYNE2/FOS/IGF1R...,25
2,GO:0010720,positive regulation of cell development,33/478,460/18986,0.071739,2.849454,6.453141,7.526254e-08,0.000086,0.000076,NEURL1/RET/EGR2/ZMIZ1/EIF4G2/ARID2/HMGB1/FOS/C...,33
3,GO:0030900,forebrain development,31/478,416/18986,0.074519,2.959879,6.495453,8.080072e-08,0.000086,0.000076,EMX2/ZMIZ1/PTEN/TH/SLC1A2/LHX5/SYNE2/FOS/IGF1R...,31
4,GO:0021543,pallium development,20/478,195/18986,0.102564,4.073812,6.933606,1.145785e-07,0.000098,0.000087,EMX2/ZMIZ1/PTEN/TH/LHX5/SYNE2/FOS/IGF1R/NDE1/F...,20
...,...,...,...,...,...,...,...,...,...,...,...,...
82,GO:1902414,protein localization to cell junction,10/478,120/18986,0.083333,3.309972,4.079406,9.148459e-04,0.046717,0.041265,CDH5/ASIC2/BAIAP2/PRKCZ/CACNG2/C1QL2/KIF5C/BSN...,10
83,GO:0030010,establishment of cell polarity,12/478,164/18986,0.073171,2.906317,3.940249,9.357090e-04,0.046717,0.041265,IGF1R/NDE1/PKD1/CDH5/STK11/PRKCZ/KIF26B/SH3BP1...,12
84,GO:0071229,cellular response to acid chemical,9/478,100/18986,0.090000,3.574770,4.148661,9.453921e-04,0.046717,0.041265,NEURL1/CPEB3/RPTOR/BAIAP2/CUL3/BCL11A/PDGFRA/F...,9
85,GO:0043410,positive regulation of MAPK cascade,24/478,472/18986,0.050847,2.019644,3.605030,9.462775e-04,0.046717,0.041265,RET/IGF2/FGF19/HMGB1/CCDC88C/IGF1R/TRAF7/DHX33...,24


There are several functional enrichment terms found related to asthma, but they are not in the top terms (not even in top 40):

* lung development	
* respiratory tube development
* respiratory system development
* gland development	
* cell-substrate adhesion	
* ...


## Model 3:


Too few signals (pvalues) to run functional enrichment analysis.

## Why so dramatic differences?


### Model 1 vs Model 3

Model 1 (binary GLMM LASSO):

logit(Pr(AA)) ~ methylation (transformed) + ... + (1|FID)

We are asking "how well does methylation at each CpG predict AA status?"


Model 3: 

transformed methylation ~ AA status + ... + (1|FID)

Here we are asking the question "how much does AA status explain methylation at each CpG?"


Direction matters!

A CpG that is a strong predictor of AA status Do NOT  == strong "effect" of AA status on methylation of this CpG, and vice versa.



### Model 2 vs Model 3


Model 2:  an un-penalized linear mixed‐effects model. it uses likelihood‐ratio or Wald test for the AA status coefficient. Those tests yield classical p-values under the null distribution.

Model 3: 

* I calculated the pvalue from model using approximate likelihood estimation (i.e. Fisher-scoring refit) on the LASSO-selected coefficients.

* This is a problem due to shrinkage bias:

  * The penalty shrinks coefficients toward zero—often substantially—so even true signals would get its coefficient shrunk! This shrinkage inflates standard errors and therefore make p-values large.

  * In my post-analysis diagnosis, I found that for some CpGs with weak (but significiant) signal (in model 2), the model 3 would aggresively zero out the coefficients, and lead to high pvaues.


### Conclusion

The direction of the association + the shrunk effect lead to barren results of model 3.

Th current pvalue from model 1 and 3 is not valid. 



## Next Steps



* For model 3, screen CpGs with model 3 to get short list of sites and feature sets, then refit an un-penalized mixed effect model to get valid pvalues for model 3.


* For model 1,  try similar refit and evaluate the differences afterwards.




